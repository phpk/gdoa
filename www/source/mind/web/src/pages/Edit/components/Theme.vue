<template>
  <Sidebar ref="sidebar" title="主题">
    <div class="themeList">
      <div
        class="themeItem"
        v-for="item in themeList"
        :key="item.value"
        @click="useTheme(item)"
        :class="{ active: item.value === theme }"
      >
        <div class="imgBox">
          <img :src="item.img" alt="" />
        </div>
        <div class="name">{{ item.name }}</div>
      </div>
    </div>
  </Sidebar>
</template>

<script>
import Sidebar from "./Sidebar";
import { themeList } from "simple-mind-map/src/utils/constant";
import { storeConfig } from "@/api";

/**
 * @Author: 王林
 * @Date: 2021-06-24 22:53:04
 * @Desc: 主题
 */
export default {
  name: "Theme",
  components: {
    Sidebar,
  },
  props: {
    mindMap: {
      type: Object,
    },
  },
  data() {
    return {
      themeList,
      theme: "",
    };
  },
  created() {
    this.$bus.$on("showTheme", () => {
      this.$refs.sidebar.show = false;
      this.$nextTick(() => {
        this.theme = this.mindMap.getTheme();
        this.$refs.sidebar.show = true;
      });
    });
  },
  methods: {
    /**
     * @Author: 王林
     * @Date: 2021-06-24 23:04:38
     * @Desc: 使用主题
     */
    useTheme(theme) {
      this.theme = theme.value;
      this.mindMap.setTheme(theme.value);
      storeConfig({
        theme: {
            "template": theme.value,
            "config": this.mindMap.getCustomThemeConfig()
        }
      });
    },
  },
};
</script>

<style lang="less" scoped>
.themeList {
  padding: 20px;

  .themeItem {
    width: 100%;
    cursor: pointer;
    border-bottom: 1px solid #e9e9e9;
    margin-bottom: 20px;
    padding-bottom: 20px;
    transition: all 0.2s;
    border: 1px solid transparent;

    &:last-of-type {
      border: none;
    }

    &:hover {
      box-shadow: 0 1px 2px -2px rgba(0, 0, 0, 0.16),
        0 3px 6px 0 rgba(0, 0, 0, 0.12), 0 5px 12px 4px rgba(0, 0, 0, 0.09);
    }

    &.active {
      border: 1px solid #67c23a;
    }

    .imgBox {
      width: 100%;

      img {
        width: 100%;
      }
    }
    .name {
      text-align: center;
      font-size: 14px;
    }
  }
}
</style>
