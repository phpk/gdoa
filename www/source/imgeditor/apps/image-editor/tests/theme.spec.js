import Theme from '@/ui/theme/theme';
import defaultTheme from '@/ui/theme/standard';

describe('Theme', () => {
  let theme;

  beforeEach(() => {
    theme = new Theme(defaultTheme);
  });

  describe('getStyle()', () => {
    it('should have "path" and "name" when the user sets the icon file location', () => {
      const userIconPath = 'fixtures/icon-d.svg';
      const userIconName = 'icon-d';
      const themeForIconPathSet = new Theme({
        ...defaultTheme,
        'menu.normalIcon.path': userIconPath,
        'menu.normalIcon.name': userIconName,
      });
      const {
        normal: { path, name },
      } = themeForIconPathSet.getStyle('menu.icon');

      expect(path).toBe(userIconPath);
      expect(name).toBe(userIconName);
    });

    it('should return default icon color information', () => {
      const { normal, active, disabled, hover } = theme.getStyle('menu.icon');

      expect(normal.color).toBe('#8a8a8a');
      expect(active.color).toBe('#555555');
      expect(disabled.color).toBe('#434343');
      expect(hover.color).toBe('#e9e9e9');
    });

    it('should return cssText in normal types', () => {
      theme.styles.normal = {
        backgroundColor: '#fdba3b',
        border: '1px solid #fdba3b',
        color: '#fff',
        fontFamily: 'NotoSans, sans-serif',
        fontSize: '12px',
      };

      expect(theme.getStyle('normal')).toMatchSnapshot();
    });

    it('should return cssText if all members are objects', () => {
      theme.styles['submenu.normalLabel'] = {
        color: '#858585',
        fontWeight: 'normal',
      };
      theme.styles['submenu.activeLabel'] = {
        color: '#000',
        fontWeight: 'normal',
      };

      expect(theme.getStyle('submenu.label')).toMatchSnapshot();
    });
  });

  describe('_makeCssText()', () => {
    it('should return the cssText of the expected value for the object', () => {
      const styleObject = {
        backgroundColor: '#fff',
        backgroundImage: './img/bg.png',
        border: '1px solid #ddd',
        color: '#222',
        fontFamily: 'NotoSans, sans-serif',
        fontSize: '12px',
      };

      expect(theme._makeCssText(styleObject)).toMatchSnapshot();
    });
  });

  describe('_makeSvgItem()', () => {
    it('should create path prefix and use-default class when using the default icon', () => {
      const useTagString = theme._makeSvgItem(['normal'], 'crop');

      expect(useTagString).toMatchSnapshot();
    });

    it('should create a svg path with the prefix when set the icon file', () => {
      const themeForIconPathSet = new Theme({
        ...defaultTheme,
        'menu.normalIcon.path': 'fixtures/icon-d.svg',
        'menu.normalIcon.name': 'icon-d',
      });
      const useTagString = themeForIconPathSet._makeSvgItem(['normal'], 'crop');

      expect(useTagString).toMatchSnapshot();
    });
  });
});
