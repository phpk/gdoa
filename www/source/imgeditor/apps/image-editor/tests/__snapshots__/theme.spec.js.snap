// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme _makeCssText() should return the cssText of the expected value for the object 1`] = `"background-color: #fff;background-image: url(./img/bg.png);border: 1px solid #ddd;color: #222;font-family: NotoSans, sans-serif;font-size: 12px"`;

exports[`Theme _makeSvgItem() should create a svg path with the prefix when set the icon file 1`] = `"<use xlink:href=\\"fixtures/icon-d.svg#icon-d-ic-crop\\" class=\\"normal\\"/>"`;

exports[`Theme _makeSvgItem() should create path prefix and use-default class when using the default icon 1`] = `"<use xlink:href=\\"#ic-crop\\" class=\\"normal use-default\\"/>"`;

exports[`Theme getStyle() should return cssText if all members are objects 1`] = `
Object {
  "active": "color: #000;font-weight: normal",
  "normal": "color: #858585;font-weight: normal",
}
`;

exports[`Theme getStyle() should return cssText in normal types 1`] = `"background-color: #fdba3b;border: 1px solid #fdba3b;color: #fff;font-family: NotoSans, sans-serif;font-size: 12px"`;
